redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -51,4 +57,103 @@ export class RecommendationProvider {
   ): Promise<Recommendation> {
     return total > maxDatabaseTotal ? { name: 'useSmallerKeys' } : null;
   }
+
+  /**
+   * Check logical databases recommendation
+   * @param redisClient
+   */
+  async determineLogicalDatabasesRecommendation(
+    redisClient: Redis | Cluster,
+  ): Promise<Recommendation> {
+    if (redisClient.isCluster) {
+      return null;
+    }
+    try {
+      const info = convertRedisInfoReplyToObject(
+        await redisClient.sendCommand(
+          new Command('info', ['keyspace'], { replyEncoding: 'utf8' }),
+        ) as string,
+      );
+      const keyspace = get(info, 'keyspace', {});
+      const databasesWithKeys = Object.values(keyspace).filter((db) => {
+        const { keys } = convertBulkStringsToObject(db as string, ',', '=');
+        return keys > 0;
+      });
+      return databasesWithKeys.length > 1 ? { name: 'avoidLogicalDatabases' } : null;
+    } catch (err) {
+      return null;
```

Comment:
also you could add some logs on error to track such errors

Reply:
Will add in next recommendation